### PR TITLE
defect #2448101: fixed wink-client vulnerable lib; fixed unreleased resources stream;

### PR DESCRIPTION
**Pull request by RalucaOT**
- excluded 'wink-client' library from' atlassian-plugins-osgi-testrunner'
- resolved 'Unreleased resource stream'  by ensuring that the resource is closed using a 'finally' block when the operation is finished

### CODE REVIEW

File: src/main/java/com/microfocus/octane/plugins/rest/RestConnector.java [hunk 1]
```diff
@@ -313,33 +313,39 @@ private Response retrieveHtmlResponse(HttpURLConnection con) throws IOException
...
-        //this actually takes the data from the previously decided stream (error or input) and stores it in a byte[] inside the response
-        if (inputStream != null) {
-            ByteArrayOutputStream container = new ByteArrayOutputStream();
-
-            byte[] buf = new byte[1024];
-            int read;
-            while ((read = inputStream.read(buf, 0, 1024)) > 0) {
-                container.write(buf, 0, read);
+        } finally {
+            if (inputStream != null) {
```
Comment (vladbenchea) [inline src/main/java/com/microfocus/octane/plugins/rest/RestConnector.java:339]: in try block you already have a check for inputStream != null - can't this be avoided here?

File: src/main/java/com/microfocus/octane/plugins/rest/RestConnector.java [hunk 2]
```diff
@@ -313,33 +313,39 @@ private Response retrieveHtmlResponse(HttpURLConnection con) throws IOException
...
-            inputStream = con.getInputStream();
-        } catch (Exception e) {
+            inputStream = (con.getResponseCode() < HttpURLConnection.HTTP_BAD_REQUEST) ? con.getInputStream() : con.getErrorStream();
+            //this actually takes the data from the previously decided stream (error or input) and stores it in a byte[] inside the response
+            if (inputStream != null) {
+                ByteArrayOutputStream container = new ByteArrayOutputStream();
+
+                byte[] buf = new byte[1024];
+                int read;
+                while ((read = inputStream.read(buf, 0, 1024)) > 0) {
```
Comment (vladbenchea) [inline src/main/java/com/microfocus/octane/plugins/rest/RestConnector.java:326]: C: new line before while block

File: src/main/java/com/microfocus/octane/plugins/rest/RestConnector.java [hunk 3]
```diff
@@ -313,33 +313,39 @@ private Response retrieveHtmlResponse(HttpURLConnection con) throws IOException
...
         //select the source of the input bytes, first try "regular" input
         try {
-            inputStream = con.getInputStream();
-        } catch (Exception e) {
+            inputStream = (con.getResponseCode() < HttpURLConnection.HTTP_BAD_REQUEST) ? con.getInputStream() : con.getErrorStream();
+            //this actually takes the data from the previously decided stream (error or input) and stores it in a byte[] inside the response
+            if (inputStream != null) {
+                ByteArrayOutputStream container = new ByteArrayOutputStream();
+
+                byte[] buf = new byte[1024];
```
Comment (vladbenchea) [inline src/main/java/com/microfocus/octane/plugins/rest/RestConnector.java:324]: C: add full name to variable - buffer

File: src/main/java/com/microfocus/octane/plugins/rest/RestConnector.java (hunk 1 above)
Comment (pikachugb) [reply src/main/java/com/microfocus/octane/plugins/rest/RestConnector.java:339]: you can use IOUtils.closequietly to not use !=null here

**Review by pikachugb — APPROVED** [review]

**Pull request merged**